plenum/server/request_handlers/nym_handler.py

```diff
@@ -1,39 +1,102 @@
+        self._steward_count = 0
 
     def static_validation(self, request: Request):
         pass
 
-    def dynamic_validation(self, req: Request):
-        origin = req.identifier
+    def dynamic_validation(self, request: Request):
+        self._validate_request_type(request)
+        identifier, req_id, operation = get_request_data(request)
         error = None
-        if not self.isSteward(self.state,
-                              origin, isCommitted=False):
+        if not is_steward(self.state,
+                          identifier, is_committed=False):
             error = "Only Steward is allowed to do these transactions"
-        if req.operation.get(ROLE) == STEWARD:
-            if self.stewardThresholdExceeded(self.config):
+        if operation.get(ROLE) == STEWARD:
+            if self._steward_threshold_exceeded(self.config):
                 error = "New stewards cannot be added by other stewards " \
                         "as there are already {} stewards in the system". \
                     format(self.config.stewardThreshold)
         if error:
-            raise UnauthorizedClientRequest(req.identifier,
-                                            req.reqId,
+            raise UnauthorizedClientRequest(identifier,
+                                            req_id,
                                             error)
 
-    def apply_request(self, request: Request, batch_ts):
-        pass
+    def gen_txn_path(self, txn):
+        nym = get_payload_data(txn).get(TARGET_NYM)
+        return hexlify(nym_to_state_key(nym)).decode()
 
-    def revert_request(self, request: Request, batch_ts):
-        pass
+    def _update_state_with_single_txn(self, txn, is_committed=True):
+        self._validate_txn_type(txn)
+        nym = get_payload_data(txn).get(TARGET_NYM)
+        existing_data = get_nym_details(self.state, nym,
+                                        is_committed=is_committed)
+        txn_data = get_payload_data(txn)
+        new_data = {}
+        if not existing_data:
+            # New nym being added to state, set the TrustAnchor
+            new_data[f.IDENTIFIER.nm] = get_from(txn)
+            # New nym being added to state, set the role and verkey to None, this makes
+            # the state data always have a value for `role` and `verkey` since we allow
+            # clients to omit specifying `role` and `verkey` in the request consider a
+            # default value of None
+            new_data[ROLE] = None
+            new_data[VERKEY] = None
+
+        if ROLE in txn_data:
+            new_data[ROLE] = txn_data[ROLE]
+        if VERKEY in txn_data:
+            new_data[VERKEY] = txn_data[VERKEY]
+        new_data[F.seqNo.name] = get_seq_no(txn)
+        new_data[TXN_TIME] = get_txn_time(txn)
+        self.__update_steward_count(new_data, existing_data)
+        existing_data.update(new_data)
+        val = self.state_serializer.serialize(existing_data)
+        key = nym_to_state_key(nym)
```

Comment:
`nym_to_state_key` is the same as `self.gen_txn_path`, right? So please use `gen_txn_path`

Reply:
fixed